identity-credential/src/credential/revocation_bitmap_status.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use identity_core::common::Object;
+use identity_core::common::Url;
+use identity_core::common::Value;
+use identity_did::did::DIDUrl;
+use identity_did::did::DID;
+
+use super::Status;
+use crate::error::Error;
+use crate::error::Result;
+
+/// Information used to determine the current status of a [`Credential`][crate::credential::Credential].
+#[derive(Clone, Debug, PartialEq)]
+pub struct RevocationBitmapStatus(Status);
```

Comment:
This could use the now-generic properties on `Status` to make things a bit easier? Or does it complicate things?

```suggestion
struct RevocationBitmapProperties {
  #[serde(rename = "revocationBitmapIndex")]
  pub revocation_bitmap_index: u32,
  #[serde(flatten)]
  pub properties: Object,
}

#[derive(Clone, Debug, PartialEq)]
pub struct RevocationBitmapStatus(Status<RevocationBitmapProperties>);
```

Reply:
Then the constructor takes a `Status<Object>` but needs to transform it to a `Status<RevocationBitmapProperties>`, and that's not easily possible without going through `serde_json::Value`, or is it?

Reply:
The constructor would use `Status::new_with_properties` instead, passing a new `RevocationBitmapProperties` to it. Any conversion from `Status<RevocationBitmapProperties>` to `Status<Object>` might need to go through `serde` though?

Probably not worth the hassle, I don't think this would affect developers much, I think it's mostly internal.